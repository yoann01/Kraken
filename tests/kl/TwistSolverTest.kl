require Math;
require Kraken;

operator entry() {

  TwistSolver solver();

  Scalar bias = 0.75;

  Mat44 origin;
  origin.setTranslation(Vec3(0, 0, 0));
  Mat44 originUpV;
  originUpV.setTranslation(Vec3(0, 1, 0));

  originUpV = origin * originUpV;

  Quat insertOri;
  insertOri.setFromAxisAndAngle(Vec3(1, 0, 0), Math_degToRad(90));

  Mat44 insert;
  insert.setTranslation(Vec3(5, 0, 0));
  insert.setRotation(insertOri);

  Mat44 insertUpV;
  insertUpV.setTranslation(Vec3(0, 1, 0));

  insertUpV = insert * insertUpV;

  Mat44 outPose[];
  outPose.resize(5);

  solver.solve(false, Scalar(1.0), bias, origin, originUpV, insert, insertUpV, outPose);

  for(Integer i=0; i < outPose.size(); i++) {
    if (i > 0) {
      report('');
    }

    report('Pose: ' + i);
    report('Translation: ' + outPose[i].translation());
    report('X Rotation:  ' + Math_radToDeg(Xfo(outPose[i]).ori.toEulerAngles().x));
    report('Scale:       ' + Xfo(outPose[i]).sc);
  }

}
