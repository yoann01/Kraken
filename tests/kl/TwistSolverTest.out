[FABRIC:MT] Ignored extension directory 'D:\fabric\FabricEngine-2.1.0-Windows-x86_64\Tests\Exts': unable to open directory 'D:\fabric\FabricEngine-2.1.0-Windows-x86_64\Tests\Exts': The system cannot find the path specified. (code 3 [0x3])Pose: 0Translation: {x:+0.0,y:+0.0,z:+0.0}X Rotation:  +0.0Scale:       {x:+1.0,y:+1.0,z:+1.0}Pose: 1Translation: {x:+1.25,y:+0.0,z:+0.0}X Rotation:  +45.0Scale:       {x:+1.0,y:+1.0,z:+1.0}Pose: 2Translation: {x:+2.5,y:+0.0,z:+0.0}X Rotation:  +67.50001Scale:       {x:+1.0,y:+1.0,z:+1.0}Pose: 3Translation: {x:+3.75,y:+0.0,z:+0.0}X Rotation:  +81.0Scale:       {x:+1.0,y:+1.0,z:+1.0}Pose: 4Translation: {x:+5.0,y:+0.0,z:+0.0}X Rotation:  +90.00001Scale:       {x:+1.0,y:+1.0,z:+1.0}